Note about file path source maps and some cleanup (#3040)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/javascript/source-maps.md

* [Sass](https://sass-lang.com/) or [PostCSS](https://postcss.org/), which can produce source maps for CSS.
* The [TypeScript](https://www.typescriptlang.org/) compiler, which compiles TypeScript to JavaScript and can produce source maps to debug the original TypeScript code.
* The [Babel](https://babeljs.io/) transpiler which can produce both CSS and JavaScript source maps.
* Build tools like [Webpack](https://webpack.js.org/), [Rollup](https://rollupjs.org/), [Vite](https://vitejs.dev/), and [Parcel](https://parceljs.org/), which can also produce source maps.

This article explains how to enable source maps in DevTools, and how to use them to debug your code. It doesn't explain how to produce source maps in your build process. To learn more about publishing source maps to the Azure Artifacts symbol server, see [Securely debug original code by publishing source maps to the Azure Artifacts symbol server](./publish-source-maps-to-azure.md).

<!-- ====================================================================== -->
## Source maps in DevTools

> Source maps from build tools cause DevTools to load your original files in addition to your minified files and replace the minified code with the original code. For example:

Reviewer: ```suggestion
Source maps from build tools cause DevTools to load your original files in addition to your minified files, and replace the minified code with the original code. For example:
```
* comma - a list of two thoughts, so separate w/ comma delimiter

* In the **Sources** tool, you can see the original files and set breakpoints in them.
* In the **Performance** tool, you can see your original function names in the flame chart.
* In the **Console** tool, you can see your original file names and line numbers in stack traces.

> Meanwhile, Microsoft Edge actually runs your minified code to render the webpage. Source maps are only used by DevTools, and only for display purposes.

Reviewer: ```suggestion
Meanwhile, Microsoft Edge actually runs your minified code to render the webpage. Source maps are only used by DevTools, and only for displaying source code to developers.
```

## Enable source maps in DevTools

Source maps are enabled by default.

To make sure that source maps are enabled:

1. To open DevTools, in Microsoft Edge, right-click a webpage, and then select **Inspect**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).

1. In DevTools, click **Customize and control DevTools** (![The Customize and control DevTools icon](./source-maps-images/customize-and-control-devtools-icon.png)) > **Settings** (![Settings icon](./source-maps-images/settings-gear-icon-light-theme.png)) > **Preferences**.

1. In the **Preferences** page, in the **Sources** section, make sure the **Enable JavaScript source maps** checkbox and the **Enable CSS source maps** checkbox are selected:

   ![The Preferences page's Sources section with the 'Enable source maps' checkboxes selected](./source-maps-images/javascript-settings-preferences-sources-enable-javascript-source-maps.png)

1. In the upper right of **Settings**, click the **Close** (![The close icon](./source-maps-images/close-icon.png)) button.

<!-- ------------------------------ -->
#### Enable loading source maps from remote file paths

> By default, DevTools doesn't load source maps when the source map URL is a remote file path, such as when the source map URL starts with `file://` and targets a file that's not on the current device. To enable loading source maps from file paths:

Reviewer: ```suggestion
By default, DevTools doesn't load source maps when the source map URL is a remote file path, such as when the source map URL starts with `file://` and targets a file that's not on the current device.

 To enable loading source maps from file paths:
```
sep. para. for lead-in for numbered steps list

1. In DevTools, click **Customize and control DevTools** (![The Customize and control DevTools icon](./source-maps-images/customize-and-control-devtools-icon.png)) > **Settings** (![Settings icon](./source-maps-images/settings-gear-icon-light-theme.png)) > **Preferences**.

> 1. In the **Preferences** page, in the **Sources** section, select the **Allow DevTools to load resources, such as source maps, from remote file paths. Disabled by default for security reasons.** checkbox.

Reviewer: ```suggestion
1. In the **Preferences** page, in the **Sources** section, select the checkbox **Allow DevTools to load resources, such as source maps, from remote file paths. Disabled by default for security reasons.**
```
